shipper.py

```diff
@@ -255,6 +283,71 @@ def session_user_entitled(cookie, min_lvl):
         return None
 
 
+# Eudat b2share
+def eudat_create_depot(base, access_token):
+    try:
+        headers = {"Content-Type": "application/json"}
+        base_url = ''.join((base, "/records/?access_token=", access_token))
+        # test md
+        d = {"titles": [{"title": "TestRest"}], "community": "e9b9792e-79fb-4b07-b6b4-b9c2bd06d095", "open_access": True, "community_specific": {}}
+        r = requests.post(base_url, data=json.dumps(d), headers=headers)
+        status_note(str(r.status_code) + " " + str(r.reason))
+        status_note('[debug] ' + str(r.json()))
+        status_note('created depot <' + r.json()['id'] + '>')
+        return str(r.json()['id'])
+    except:
+        raise
+
+
+def eudat_add_zip_to_depot(base, deposition_id, zip_name, target_path, token):
+    try:
+        fsum = files_dir_size(target_path)
+        if fsum <= env_max_dir_size_mb:
+            # get bucket url:
+            headers = {"Content-Type": "application/json"}
+            r = requests.get(''.join((base, '/records/', deposition_id, '/draft?access_token=', token)), headers=headers)
+            bucket_url = r.json()['links']['files']
+            if r.status_code == 200:
+                status_note(str(r.status_code) + ' using bucket <' + bucket_url + '>')
+            else:
+                status_note(r.status_code)
+            # upload file into bucket:
+            headers = {"Content-Type": "application/octet-stream"}
+            # create a filelike object in memory
+            filelike = BytesIO()
```

Comment:
we already had a similar segment in for zipping when it was only zenodo :)
I thinks it's a good way to stream the zip file right away during creation, however, it still needs to prove robustness. anyway this works while we move in the gigabytes...